lib/trie/trie.go

```diff
@@ -353,14 +357,24 @@ func (t *Trie) updateBranch(p *branch, key []byte, value node) (n node) {
 		switch c := p.children[key[length]].(type) {
 		case *branch, *leaf:
 			n = t.insert(c, key[length+1:], value)
-			p.children[key[length]] = n
+
+			cpy := [16]node{}
+			copy(cpy[:], p.children[:])
+			cpy[key[length]] = n
+			p.children = cpy
```

Comment:
What's the point here if we set back `cpy` to `p.children`? :thinking: Same question for below.

Reply:
I don't think this is actually needed since children is an array :p